src/pages/resultsView/enrichments/EnrichmentsUtil.tsx

```diff
@@ -408,22 +408,25 @@ export function getGroupColumns(groups: { name: string, description: string }[],
     return columns;
 }
 
-export function getEnrichmentBarPlotData(data: AlterationEnrichmentRow[], genes: string[]): IMultipleCategoryBarPlotData[] {
-    const usedGenes: {[gene:string]:boolean} = {};
+export function getEnrichmentBarPlotData(data: { [gene: string]: AlterationEnrichmentRow }, genes: string[]): IMultipleCategoryBarPlotData[] {
+    const usedGenes: { [gene: string]: boolean } = {};
     if (_.isEmpty(genes)) {
         return [];
     }
-    let groupToGeneCounts = _.reduce(data, (acc, datum) => {
-        _.each(datum.groupsSet, group => {
-            const groupName = group.name + (datum.qValue < 0.05 ? '*' : '')
-            if (genes.includes(datum.hugoGeneSymbol)) {
+
+    const groupToGeneCounts = _.reduce(genes, (acc, gene) => {
+        const datum = data[gene];
+        if (datum) {
+            _.each(datum.groupsSet, group => {
+                const groupName = group.name;
                 if (!acc[groupName]) {
                     acc[groupName] = {};
                 }
-                acc[groupName][datum.hugoGeneSymbol] = group.alteredCount;
-                usedGenes[datum.hugoGeneSymbol] = true;
-            }
-        });
+                const displayedGene = gene + (datum.qValue < 0.05 ? '*' : '')
+                acc[groupName][displayedGene] = group.alteredPercentage;
```

Comment:
is there any way you could add the * at render time?  or make a property of this object called displayedGeneName, rather than use the asterisked name as a key?

Reply:
this would involve lot of changes `MultipleCategoryBarPlot` component.